handler/src/main/java/io/netty/handler/ssl/SslContext.java

```diff
@@ -673,6 +673,204 @@ static ApplicationProtocolConfig toApplicationProtocolConfig(Iterable<String> ne
         return apn;
     }
 
+    /**
+     * Creates a builder for new client-side {@link SslContext}.
+     */
+    public static Builder newBuilderForClient() {
+        return new Builder(true);
+    }
+
+    /**
+     * Creates a builder for new server-side {@link SslContext}.
+     *
+     * @param keyCertChainFile an X.509 certificate chain file in PEM format
+     * @param keyFile a PKCS#8 private key file in PEM format
+     */
+    public static Builder newBuilderForServer(File keyCertChainFile, File keyFile) {
+        return new Builder(false).keyManager(keyCertChainFile, keyFile);
+    }
+
+    /**
+     * Creates a builder for new server-side {@link SslContext}.
+     *
+     * @param keyCertChainFile an X.509 certificate chain file in PEM format
+     * @param keyFile a PKCS#8 private key file in PEM format
+     * @param keyPassword the password of the {@code keyFile}, or {@code null} if it's not
+     *     password-protected
+     */
+    public static Builder newBuilderForServer(File keyCertChainFile, File keyFile,
+            String keyPassword) {
+        return new Builder(false).keyManager(keyCertChainFile, keyFile, keyPassword);
+    }
+
+    /** Builder for configuring a new SslContext for creation. */
+    public static final class Builder {
```

Comment:
I was talking about the static factory methods for the Builder itself. Such factory methods are useful to give _names_ to the different use cases of creating the SslContextBuilder. Doing ```new SslContextBuilder(true)``` isn't very obvious. We could make it such that the server factory methods don't have any arguments, but it seems nice that the builder encourages specifying the required parameters instead of making people read the documentation more carefully.

I wasn't referring to optional things like trust manager and mutual auth; those are simply specified on the builder with normal methods after builder construction.